File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        if x >= 0:
            x = int(str(x)[::-1])
        else:
            x = -int(str(x)[::-1][:-1])
        if abs(x) > 2**31:
            return 0
        else:
            return x

File: 0007-reverse-integer/0007-reverse-integer.java
import java.lang.Math;
class Solution {
    public int reverse(int x) {
        int y = 0;
        while(x != 0){
            if(Math.abs(y) > Math.pow(2, 31) / 10){
                return 0;
            }
            y =  y * 10 + x % 10;
            x /= 10;
        }
        return y;
    }
}